src/WebJobs.Script/Workers/Rpc/Configuration/RpcWorkerConfigFactory.cs

```diff
@@ -159,6 +180,44 @@ internal void AddProvider(string workerDir)
             }
         }
 
+        private List<WorkerDescriptionProfile> ReadWorkerDescriptionProfiles(JToken profilesJToken)
+        {
+            var profiles = profilesJToken.ToObject<IList<WorkerProfileDescriptor>>();
+
+            if (profiles == null || profiles.Count <= 0)
+            {
+                return new List<WorkerDescriptionProfile>(0);
+            }
+
+            var descriptionProfiles = new List<WorkerDescriptionProfile>(profiles.Count);
+            try
+            {
+                foreach (var profile in profiles)
+                {
+                    var profileConditions = new List<IWorkerProfileCondition>(profile.Conditions.Count);
+
+                    foreach (var descriptor in profile.Conditions)
+                    {
+                        if (!_profileManager.TryCreateWorkerProfileCondition(descriptor, out IWorkerProfileCondition condition))
+                        {
+                            // Failed to resolve condition. This profile will be disabled using a mock false condition
+                            _logger?.LogInformation($"Profile {profile.Name} is disabled. Cannout resolve the profile condition {descriptor.Type}");
+                            condition = new FalseCondition();
+                        }
+
+                        profileConditions.Add(condition);
+                    }
+
+                    descriptionProfiles.Add(new (profile.Name, profileConditions, profile.Description));
+                }
+            }
+            catch (Exception)
+            {
+                throw new FormatException("Failed to parse profiles in worker config.");
```

Comment:
Maybe create a detector for this

Reply:
I think we should definitly create a dectector for profiles to know if we:

a) loaded a profile
b) tried to load a profile but failed
c) a profile failed validation
d) found profiles in the worker.config but none of the conditions were met (and so did not load)